tests/topics/test_topic_list_endpoint.py

```diff
@@ -0,0 +1,176 @@
+
+    assert result.status_code == 400
+    assert result.json == {"message": "empty argument value is not allowed"}
+
+
+def test_get_topic_list_endpoint_with_invalid_author_id_returns_400_correct_body(client, db_with_one_user):
+    authorization_token = jwt.encode({"user_id": 1}, Config.SECRET_KEY, algorithm="HS256")
+
+    result = client.get("api/topics", headers={"Authorization": f"Bearer {authorization_token}"},
+                        query_string={"author_id": "InvalidArgument"})
+
+    assert result.status_code == 400
+    assert result.json == {"message": "author id must be an integer"}
+
+
+def test_get_topic_list_endpoint_with_created_before_returns_200(client, db_with_three_users_and_three_topics):
+    session = db_with_three_users_and_three_topics
+    authorization_token = jwt.encode({"user_id": 1}, Config.SECRET_KEY, algorithm="HS256")
+    oldest_topic_creation_date = session.query(Topic).filter_by(id=3).one().created_at
+    filtered_topic_list = session.query(Topic).filter(Topic.created_at < oldest_topic_creation_date).all()
+
+    result = client.get("api/topics", headers={"Authorization": f"Bearer {authorization_token}"},
+                        query_string={"created_before": str(oldest_topic_creation_date)})
+
+    assert result.status_code == 200
+    assert result.json == [{
+        "id": topic.id,
+        "author_id": topic.author_id,
+        "created_at": str(topic.created_at),
+        "description": topic.description,
+        "title": topic.title,
+    } for topic in filtered_topic_list]
+
+
+def test_get_topic_list_endpoint_with_empty_created_before_returns_400_correct_body(client, db_with_one_user):
+    authorization_token = jwt.encode({"user_id": 1}, Config.SECRET_KEY, algorithm="HS256")
+
+    result = client.get("api/topics", headers={"Authorization": f"Bearer {authorization_token}"},
+                        query_string={"created_before": ""})
+
+    assert result.status_code == 400
+    assert result.json == {"message": "empty argument value is not allowed"}
+
+
+def test_get_topic_list_endpoint_with_invalid_created_before_returns_400_correct_body(client, db_with_one_user):
+    authorization_token = jwt.encode({"user_id": 1}, Config.SECRET_KEY, algorithm="HS256")
+
+    result = client.get("api/topics", headers={"Authorization": f"Bearer {authorization_token}"},
+                        query_string={"created_before": "InvalidArgument"})
+
+    assert result.status_code == 400
+    assert result.json == {"message": "invalid datetime string"}
+
+
+def test_post_topic_list_endpoint_with_valid_args_returns_200_correct_body(client, db_with_one_user):
+    session = db_with_one_user
+    authorization_token = jwt.encode({"user_id": 1}, Config.SECRET_KEY, algorithm="HS256")
+    request = client.post("api/topics", headers={"Authorization": f"Bearer {authorization_token}"},
+                          query_string={"title": "test", "description": "testtest"})
+
+    result = session.query(Topic).first()
```

Comment:
We should test POST request to /api/topics here instead of sqlalchemy's `query()` method.

Since this endpoint has side effects (storing data to database) we need to write two tests for this endpoint. One test should check the correctness of the response and second should check the correctness of the database state after update.

So will have these two tests:
```python
def test_post_list_endpoint_with_valid_args_returns_200_with_correct_body():
    result = client.post( ... )

    assert result.status_code == 200
    assert result.json == { ... }


def test_post_topic_list_endpoint_with_valid_args_creates_objects_in_db_correctly():
    result = client.post( ... )

    users = session.query(User).all()
    assert len(users) == 1
    created_user = users[0]
    assert created_user.username == ...
    assert created_user.password_hash == ...
    ...
```